R/lib.R

```diff
@@ -281,6 +281,35 @@ get_proteomics_data <- function(id) {
     dplyr::rename_all(tolower)
 }
 
+#' Get and process metabolomics. For stats,
+#' - convert list into dataframe
+#' - make id column into factor
+#' - transpose boxplot stats
+#' - trim extraneous columns
+#' Then, join processed stats and gene associations.
+#'
+#' @export
+process_metabolomics <- function(id) {
+  env <- environment()
+  synGet(id)$path %>% load(envir = env)
+
+  is.desired.shape <- function(x) if(identical(dim(x),c(2,5))) return(TRUE) else return(FALSE)
+
+  metabolomics_stats <- agora.metabolite.stats %>%
+    do.call(rbind, .) %>%
+    as.data.frame(stringsAsFactors = FALSE) %>%
+    assertr::verify(assertr::is_uniq(metabolite.id)) %>%
+    dplyr::mutate(metabolite.id=factor(unlist(metabolite.id))) %>%
+    dplyr::mutate(boxplot.stats=map(boxplot.stats,unlist)) %>%
+    dplyr::mutate(transposed.boxplot.stats=lapply(boxplot.stats, t)) %>%
+    # assertr::assert_rows(is.desired.shape,transposed.boxplot.stats) %>%
```

Comment:
@kdaily Could not get this assertion to work. I don't think we really need this, but I would prefer to have a test for each transformation. The point of this one is to verify the dim goes from c(5,2) to c(2,5). Any tips?

Reply:
Ah! I think you want to use the `assertthat` library instead, because you want to test something about the whole data frame `metabolomics_stats`.

Reply:
thanks for the tip!